config/config.go

```diff
@@ -59,10 +72,58 @@ func Init(flags *pflag.FlagSet) {
 		if flag.Name == "help" {
 			return
 		}
-		viper.Set(flag.Name, flag.Value.String())
+
+		// Only updated parsed configs if the user sets value
+		if flag.Changed {
+			viper.Set(flag.Name, flag.Value.String())
+		}
 	})
 
 	if err := viper.Unmarshal(&Config); err != nil {
 		panic(err)
 	}
 }
+
+// ConfigureParentDirPaths Creates the default parent directory which can be used for logs and persistent data
+func configureParentDirPaths() {
+	var err error
+	DefaultParentDir, err = createDefaultParentDir()
+	if err != nil {
+		slog.Warn("Failed to create default preferences directory", slog.String("error", err.Error()))
+	}
+}
+
+func createDefaultParentDir() (string, error) {
+	// Get the default directory path for the OS
+	prefDir := GetDefaultParentDirPath()
+	if err := os.MkdirAll(prefDir, os.ModePerm); err != nil {
+		return "", fmt.Errorf("failed to create preferences directory: %w", err)
+	}
+
+	return prefDir, nil
+}
+
+func GetDefaultParentDirPath() string {
+	switch runtime.GOOS {
+	case "linux":
+		return linuxPath
+	case "darwin":
+		return expandHomeDirDarwin(darwinPath)
+	case "windows":
+		return windowsPath
+	default:
+		slog.Warn("unsupported OS, defaulting to current directory")
+		return DefaultParentDir
```

Comment:
There should not be any default. The same variable is defined across three files and that same var is ref in constants and then used here. I do not see we need this function at all after the above changes.